GVFS/GVFS.Service/ProductUpgradeTimer.cs

```diff
@@ -20,15 +20,21 @@ public ProductUpgradeTimer(JsonTracer tracer)
 
         public void Start()
         {
-            Random random = new Random();
-            TimeSpan startTime = TimeSpan.Zero;
+            if (!GVFSEnlistment.IsUnattended(this.tracer))

```

Comment:
@alameenshah rookie question here, what does it mean for an enlistment to unattended?

Reply:
It means VFSForGit is running un-attended by any user, may be inside a lab. This state can be enabled by setting the environment variable ```GVFS_UNATTENDED```.